block/manager_test.go

```diff
@@ -280,3 +281,83 @@ func getTempKVStore(t *testing.T) ds.TxnDatastore {
 	require.NoError(t, err)
 	return kvStore
 }
+
+func Test_isProposer(t *testing.T) {
+	require := require.New(t)
+
+	type args struct {
+		genesis       *cmtypes.GenesisDoc
+		signerPrivKey crypto.PrivKey
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "Signing key matches genesis proposer public key",
+			args: func() args {
+				genesisData, privKey := types.GetGenesisWithPrivkey()
+				signingKey, err := types.PrivKeyToSigningKey(privKey)
+				require.NoError(err)
+				return args{
+					genesisData,
+					signingKey,
+				}
+			}(),
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "Signing key does not match genesis proposer public key",
+			args: func() args {
+				genesisData, _ := types.GetGenesisWithPrivkey()
+				randomPrivKey := ed25519.GenPrivKey()
+				signingKey, err := types.PrivKeyToSigningKey(randomPrivKey)
+				require.NoError(err)
+				return args{
+					genesisData,
+					signingKey,
+				}
+			}(),
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "No validators found in genesis",
+			args: func() args {
+				genesisData, privKey := types.GetGenesisWithPrivkey()
+				genesisData.Validators = nil
+				signingKey, err := types.PrivKeyToSigningKey(privKey)
+				require.NoError(err)
+				return args{
+					genesisData,
+					signingKey,
+				}
+			}(),
+			want:    false,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := isProposer(tt.args.genesis, tt.args.signerPrivKey)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("isProposer() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("isProposer() = %v, want %v", got, tt.want)
+			}
```

Comment:
then this can be updated to be

```suggestion
			isProposer, err := isProposer(tt.args.genesis, tt.args.signerPrivKey)
			if err != tt.err {
				t.Errorf("isProposer() error = %v, expected err %v", err, tt.err)
				return
			}
			if isProposer != tt.isProposer {
				t.Errorf("isProposer() = %v, expected %v", isProposer, tt.isProposer)
			}
```

Reply:
Updated